apps/emails-and-messages/src/modules/sendgrid/configuration/sendgrid-config-container.ts

```diff
@@ -55,14 +61,28 @@ const getConfigurations =
 
     let filtered = sendgridConfigRoot.configurations;
 
-    if (filter?.ids?.length) {
+    if (!filter) {
+      return filtered;
+    }
+
+    if (filter.ids?.length) {
       filtered = filtered.filter((c) => filter?.ids?.includes(c.id));
     }
 
-    if (filter?.active !== undefined) {
+    if (filter.active !== undefined) {
       filtered = filtered.filter((c) => c.active === filter.active);
     }
 
+    if (filter.availableInChannel?.length) {
+      filtered = filtered.filter((c) =>
+        isAvailableInChannel({
+          channel: filter.availableInChannel!,
+          restrictedToChannels: c.channels.restrictedTo,
+          excludedChannels: c.channels.excludedFrom,
+        })
+      );
+    }
+
     return filtered;
```

Comment:
I think this code is duplicated - should be extracted, because it contains some shared behavior